astropy/wcs/src/wcslib_auxprm_wrap.c

```diff
@@ -0,0 +1,234 @@
+#define NO_IMPORT_ARRAY
+
+#include "astropy_wcs/wcslib_auxprm_wrap.h"
+
+#include <wcs.h>
+#include <wcsprintf.h>
+#include <tab.h>
+
+/*
+ It gets to be really tedious to type long docstrings in ANSI C syntax
+ (since multi-line strings literals are not valid).  Therefore, the
+ docstrings are written in doc/docstrings.py, which are then converted
+ by setup.py into docstrings.h, which we include here.
+*/
+#include "astropy_wcs/docstrings.h"
+
+
+/***************************************************************************
+ * PyAuxprm methods                                                        *
+ ***************************************************************************/
+
+static PyObject*
+PyAuxprm_new(PyTypeObject* type, PyObject* args, PyObject* kwds) {
+  PyAuxprm* self;
+  self = (PyAuxprm*)type->tp_alloc(type, 0);
+  return (PyObject*)self;
+}
+
+
+static int
+PyAuxprm_traverse(PyAuxprm* self, visitproc visit, void *arg) {
+  Py_VISIT(self->owner);
+  return 0;
+}
+
+
+static int
+PyAuxprm_clear(PyAuxprm* self) {
+  Py_CLEAR(self->owner);
+  return 0;
+}
+
+
+static void PyAuxprm_dealloc(PyAuxprm* self) {
+  PyAuxprm_clear(self);
+  Py_TYPE(self)->tp_free((PyObject*)self);
+}
+
+
+PyAuxprm* PyAuxprm_cnew(PyObject* wcsprm, struct auxprm* x) {
+  PyAuxprm* self;
+  self = (PyAuxprm*)(&PyAuxprmType)->tp_alloc(&PyAuxprmType, 0);
+  if (self == NULL) return NULL;
+  self->x = x;
+  Py_INCREF(wcsprm);
+  self->owner = wcsprm;
+  return self;
+}
+
+
+static void auxprmprt(const struct auxprm *aux) {
+
+  if (aux == 0x0) return;
+
+  if (aux->rsun_ref != UNDEFINED) wcsprintf("rsun_ref: %f\n", aux->rsun_ref);
+  if (aux->dsun_obs != UNDEFINED) wcsprintf("dsun_obs: %f\n", aux->dsun_obs);
+  if (aux->crln_obs != UNDEFINED) wcsprintf("crln_obs: %f\n", aux->crln_obs);
+  if (aux->hgln_obs != UNDEFINED) wcsprintf("hgln_obs: %f\n", aux->hgln_obs);
+  if (aux->hglt_obs != UNDEFINED) wcsprintf("hglt_obs: %f\n", aux->hglt_obs);
+
+  return;
+}
+
+
+static PyObject* PyAuxprm___str__(PyAuxprm* self) {
```

Comment:
Could we also expose this as part of the `__repr__`?

Reply:
I would not oppose to this *only if* this does not become a template for all other structures in `Wcsprm`. Specifically, implementing similar behavior for `__repr__` for `Tabprm` and `Wtbarr` would be an issue because `tab` and `wtb` members of `Wcsprm` are lists of `Tabprm` and `Wtbarr` and so doing `__repr__(tab)` would result in an a very long printout. So, as long as this is implemented *exclusively for `aux`*, I am OK.

Reply:
We do also have a long repr for `wcs.wcs`, could we have potentially have some truncated version for tab or similar?

Reply:
What should the repr look like?